File: kernel/kernel_union/src/main/java/com/ycr/kernel/union/helper/ContextHelper.java
package com.ycr.kernel.union.helper;

import android.app.Application;
import android.content.Context;
import android.content.res.Resources;
import android.support.v4.content.ContextCompat;

/**
 * created by yuchengren on 2019/4/22
 */
public class ContextHelper {
    private static Context context;
    private static Application application;

    public static Context getContext(){
        return context;
    }

    public static Application getApplication(){
        return application;
    }

    public static void  doInit(Application application){
        ContextHelper.application = application;
        context = application.getApplicationContext();
    }

    public static Resources getResources(){
        return context.getResources();
    }

    public static String getString(int resId,Object... args){
        return context.getString(resId,args);
    }

    public static int getColor(int resId){
        return ContextCompat.getColor(context,resId);
    }

    public static int getDimenPixel(int resId){
        return getResources().getDimensionPixelSize(resId);
    }

}

File: kernel/kernel_union/src/main/java/com/ycr/kernel/union/helper/ContextHelper.kt
package com.ycr.kernel.union.helper

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context
import android.content.res.ColorStateList
import android.content.res.Resources
import android.support.v4.content.ContextCompat

/**
 * created by yuchengren on 2019/4/22
 */
@SuppressLint("StaticFieldLeak")
object ContextHelper {
    private lateinit var context: Context
    private lateinit var application: Application
    val resources: Resources
        get() = context.resources

    @JvmStatic fun doInit(application: Application) {
        this.application = application
        context = application.applicationContext
    }

    @JvmStatic fun getContext(): Context{
        return context
    }

    @JvmStatic fun getApplication(): Application{
        return application
    }

    @JvmStatic fun getString(resId: Int, vararg args: Any): String {
        return context.getString(resId, *args)
    }

    @JvmStatic fun getColor(resId: Int): Int {
        return ContextCompat.getColor(context, resId)
    }

    @JvmStatic fun getColorStateList(resId: Int): ColorStateList? {
        return ContextCompat.getColorStateList(context, resId)
    }

    @JvmStatic fun getDimenPixel(resId: Int): Int {
        return resources.getDimensionPixelSize(resId)
    }
}
